### ISSUE TITLE
Ability to disable / blacklist metrics

### ISSUE THREAD

**Issue by ggarcia-te** [opstree/druid-exporter#117]
When we run Druid batch ingestions (index_parallel task and its subtasks) there can be tens of thousands of tasks created in the cluster. 

This eventually becomes a problem in our `druid-exporter` instance due to this metric:
```
druid_tasks_duration{task_id="coordinator-issued_compact_agent_ping_1m_enriched_knnnkdkj_2021-06-29T05:26:11.462Z",task_status="SUCCESS"} 442223
```

Because it exports `task_id`, which has a very high cardinality, the exporter will end up exposing 1 metric per task created. This eventually brings down the exporter or causes timeout problems when prometheus scrapes the `/metrics` endpoint in the exporter.

We would like to have the ability to disable metrics that are [pulled from Druid](https://github.com/opstree/druid-exporter/blob/0da9c52d429cf7bdfcd5b200a221606d257ff86c/collector/druid.go#L169-L222) by the exporter via http request (like tasks). Ideally, having a program argument / env variable we can use to WHITELIST / BLACKLIST metrics. ie:

```
--blacklist-metrics "DruidTasks,DruidRunningTasks,DruidWaitingTasks,DruidCompletedTasks,DruidPendingTasks"

BLACKLIST_METRICS="DruidTasks,DruidRunningTasks,DruidWaitingTasks,DruidCompletedTasks,DruidPendingTasks"
```

**Comment by acherla** [comment]
Honestly, modifying the statement below to be optional would pretty much resolve the issue.  This issue will affect situations where you have long running kafka streaming ingest jobs as well, since a large number of tasks is auto generated by druid for each task ingest it runs for kafka.  I dont think the design for pulling in the task list was particularly well designed for being exported to prometheus anyways.  

```
for _, data := range GetDruidTasksData(tasksURL) {
		hostname := ""
		for _, worker := range workers {
			for _, task := range worker.RunningTasks {
				if task == data.ID {
					hostname = worker.hostname()
					break
				}
			}
			if hostname != "" {
				break
			}
		}
		if hostname == "" {
			if len(workers) != 0 {
				hostname = workers[rand.Intn(len(workers))].hostname()
			}
		}
		ch <- prometheus.MustNewConstMetric(collector.DruidTasks,
			prometheus.GaugeValue, data.Duration, hostname, data.DataSource, data.ID, data.GroupID, data.Status, data.CreatedTime)
	}
```

**Comment by layoaster** [comment]
This is really annoying I have 3 Kafka supervisors running continuously spawning new tasks by the hour so I have to restart druid-exporter every 24h because of this issue. 

Not really experienced in Prometheus metrics but I think the task's metrics collection strategy would benefit from only collecting data for tasks which `status="RUNNING"` in the previous scrape and in the current scrape)

**Comment by T3pco** [comment]
@layoaster: as druid itself is pushing the data to the exporter (with its own interval configuration)- this would be hard to implement. Prometheus scraping interval is also independend from everything else.
To avoid continously increase of presented metrics to prometheus, there is the TTL in druid-exporter which is 5min.

I think the druid-exporter is just overloaded with huge amount of data. We have also sometimes outages of druid-epxorter when we have druid-cluster issues and alot of metrics will be generated.

Too possible solutions for you:
1. If you are fine with deleting finished tasks in druid metadata, set
druid.indexer.storage.recentlyFinishedThreshold=PT15M
in your coordinator config

This will restrict the pushed data to druid-exporter as "non-running/waiting" tasks will be deleted after 15m (or other value).
Sadly, this way also "FAILED" tasks get deleted from druid metadata.

When you scrape the task data within those 15m, you have everything stored with status in prometheus.
We managed to stabilize the druid-exporter this way.

2. If you are not really interessted in detailed task data in prometheus. Check the official prometheus support included since v0.22.0

These metrics are "designed" for prometheus - and might be more stable
https://druid.apache.org/docs/0.22.0/development/extensions-contrib/prometheus.html
